fluentd/lib/filter_elasticsearch_genid_ext/lib/filter_elasticsearch_genid_ext.rb

```diff
@@ -0,0 +1,45 @@
+require 'securerandom'
+require 'base64'
+require 'fluent/filter'
+
+module Fluent
+  class ElasticsearchGenidExtFilter < Filter
+    Fluent::Plugin.register_filter('elasticsearch_genid_ext', self)
+
+    config_param :hash_id_key, :string, :default => '_hash'
+    config_param :alt_key, :string, default: ""
+
+    def initialize
+      super
+    end
+
+    def configure(conf)
+      super
+      @alt_key = conf['alt_key']
+    end
+
+    def filter(tag, time, record)
```

Comment:
Can we add a check here to only use the `@alt_key` for a specific tag.  I can imagine this could become expensive for all records when we only need to check events

Reply:
If LOG_DRIVER is json-file, the tag contains "eventrouter" in the tag, I'd think.  But journald does not have the knowledge (for now)?  Do we need to add it?

Reply:
Yes.  You'll need to add some logic to filter-retag-journal.conf to create a special tag for eventrouter e.g. something like 
```
# mark eventrouter events
rewriterule6 CONTAINER_NAME ^k8s_[^_]+_kube-eventrouter-[^_]+_ kubernetes.journal.container._kube-eventrouter_
```
That is, if the pod-name part of CONTAINER_NAME is kube-eventrouter-*, tag as kubernetes.journal.container._kube-eventrouter_

Then, you'll need to implement programmatic tag matching in this plugin.  You can probably grab the code from the viaq plugin: https://github.com/ViaQ/fluent-plugin-viaq_data_model/blob/master/lib/fluent/plugin/filter_viaq_data_model.rb#L106
and
https://github.com/ViaQ/fluent-plugin-viaq_data_model/blob/master/lib/fluent/plugin/filter_viaq_data_model.rb#L166
and
https://github.com/ViaQ/fluent-plugin-viaq_data_model/blob/master/lib/fluent/plugin/filter_viaq_data_model.rb#L371

I suggest adding a new config param:
```
      desc 'only do the alt_tag processing on records which match these tag patterns'
      config_param :alt_tag, :string
```
Then you can specify them in the config like this:
```
   @type genid-ext
   alt_tag "#{ENV['GENID_ALT_TAG'] || 'kubernetes.var.log.containers.kube-eventrouter-*.** kubernetes.journal.container._kube-eventrouter_'}"
```

Reply:
Should we consider modifying the tag to not be 'eventrouter' and calling it kuberenetes_events or something like it.  Technically, we are processing events that so happen to be produced by the event router.  Why not tag it for what it is

Reply:
@richm , regarding this suggestion...
```
# mark eventrouter events
rewriterule6 CONTAINER_NAME ^k8s_[^_]+_kube-eventrouter-[^_]+_ kubernetes.journal.container._kube-eventrouter_
```
1) could this rewriterule6 be `^k8s_[^_]+[^_]+_kube-eventrouter-[^_]+_ kubernetes.journal.container._kube-eventrouter_`?
2) if so, this existing rewriterule already does the rewrite?  (i.e., the answer to my original question about journald was "no"?)
```
# mark logs from kube-* namespaces for processing as k8s logs but stored as system logs
rewriterule3 CONTAINER_NAME ^k8s_[^_]+_[^_]+_kube-(.+)_ kubernetes.journal.container._kube-$1_
```

Reply:
> could this rewriterule6 be ^k8s_[^_]+[^_]+_kube-eventrouter-[^_]+_ kubernetes.journal.container._kube-eventrouter_?

No.  That will match a _namespace name_ that starts with "kube-eventrouter".  We want to match on the _pod name_.

> Should we consider modifying the tag to not be 'eventrouter' and calling it kuberenetes_events or something like it. Technically, we are processing events that so happen to be produced by the event router. Why not tag it for what it is

Sure.  That will work for log-driver=journald.  But for json-file or crio, we have no control over the tag, unless we want to do a retag.

Reply:
Is that expensive?  I don't know if that means there are significant change to the pipeline?

Reply:
> Is that expensive?

Do you mean doing the retag for the json-file/crio case?  Yes, in terms of both performance and complexity.  Which is why I'm reluctant to do that.

> its from journal or its from a container is not significant

It is - in both the viaq filter formatting, and the routing to figure out if it is a project/apps log or an operations/infra log.

Reply:
The problem with `kubernetes.event` is that the viaq filter plugin won't know that this is a record from the journal, and so will not add all of the journal metadata to the record . . . that is a regression from the current behavior.